common/path_manager.cpp

```diff
@@ -38,66 +34,64 @@ void PathManager::LoadPaths()
 
 	const auto c = EQEmuConfig::get();
 
-	// maps
-	if (File::Exists(fs::path{m_server_path + "/" + c->MapDir}.string())) {
-		m_maps_path = fs::relative(fs::path{m_server_path + "/" + c->MapDir}).string();
-	}
-	else if (File::Exists(fs::path{m_server_path + "/maps"}.string())) {
-		m_maps_path = fs::relative(fs::path{m_server_path + "/maps"}).string();
-	}
-	else if (File::Exists(fs::path{m_server_path + "/Maps"}.string())) {
-		m_maps_path = fs::relative(fs::path{m_server_path + "/Maps"}).string();
-	}
-
-	// quests
-	if (File::Exists(fs::path{m_server_path + "/" + c->QuestDir}.string())) {
-		m_quests_path = fs::relative(fs::path{m_server_path + "/" + c->QuestDir}).string();
+	auto resolve_path = [&](const std::string& dir, const std::vector<std::string>& fallback_dirs = {}) -> std::string {
+		// relative
+		if (File::Exists(fs::path{m_server_path + "/" + dir}.string())) {
+			return fs::relative(fs::path{m_server_path + "/" + dir}).lexically_normal().string();
+		}
+
+		// absolute
+		if (File::Exists(fs::path{dir}.string())) {
+			return fs::absolute(fs::path{dir}).string();
+		}
```

Comment:
using std::filesystem::path to join instead of string concatenation will handle both of these cases:

```
fs::path server_path = fs::path(m_server_path);
std::error_code ec;

if (fs::path dir_path = server_path / dir; fs::is_directory(dir_path, ec) {
    return fs::proximate(dir_path).lexically_normal().string()
}
```